### DISCUSSION TITLE
No wireless interfaces found

### DISCUSSION THREAD

**Discussion by rock-crusher** [tagyoureit/nodejs-poolController#733] [Nixie]
So I did a git pull on nodejs-poolController & dashPanel, now I can’t reconnect to dash panel do to no internet connection. The Wi-Fi icon now shows the double red XX and says “No wireless interfaces found” 
Everything was fine prior to the git pull, something obviously went wrong when I did this, but now I’m not sure how to correct the problem.
Any suggestions?

**Comment by rstrouse** [comment]
Did you run npm i

**Reply by rock-crusher** [reply]
Hi rstrouse!
Yes, this is exactly how I did it,

First:

cd ~/nodejs-poolController

git pull

npm i

second:



cd ~/nodejs-poolController-dashPanel

git pull

npm i

**Comment by rock-crusher** [comment]
I just can’t seem to figure out how to get the Wi-Fi working again?

Seriously, this shouldn’t be all that difficult right? 🤷‍♂️

It must be something in the settings that I’m overlooking, or a corrupt file somewhere? 😒

Can simply doing a git pull somehow foul up the system in some way?  🤔

I should have left things alone, it was working fine, but I thought what the hell, may as well get up to speed with the updates. 😢

**Reply by rstrouse** [reply]
Double check that Wi-Fi is enabled on the pi and disable Bluetooth.

**Reply by rock-crusher** [reply]
Yes, Bluetooth is disabled, but there doesn’t appear to be a wireless interface available to enable, or disable for that matter. 

I haven’t found a solution for this and beginning to wonder how exactly I screwed this up? 

This is what I’ve tried and it stops me dead in my tracks, I don’t know how to proceed any further.

Opened a terminal: “sudo raspi-config” 

This brought me to: “System Options”

Then to: “Wireless Lan”

Then to: “No wireless interface found”

Then to: “There was an error running option S1 Wireless Lan”

I simply have no idea what the next move should be.

![terminal](https://user-images.githubusercontent.com/94593954/233899928-3316d215-5c07-48e9-a805-bf14800982c8.png)
![System Options](https://user-images.githubusercontent.com/94593954/233900117-f1c01433-7092-4833-8d43-32545ae1517e.png)
![Wireless Lan](https://user-images.githubusercontent.com/94593954/233900192-7bcb0e87-e725-457a-9c3f-a114e275b63e.png)
![No wireless interface found](https://user-images.githubusercontent.com/94593954/233900217-48fcdf2e-804e-4822-8cb8-7502a433a2f7.png)
![error running option S1](https://user-images.githubusercontent.com/94593954/233900263-14eef4bd-06fc-436c-9d29-09049320aa15.png)

**Comment by rstrouse** [comment]
I do it from the UI.  On the upper right corner you will see an up/down arrow click on it and see if the menu says Turn On Wi-Fi.

**Reply by rock-crusher** [reply]
Mine has a red X covering both arrows, if I click on it “No wireless interfaces found” is what it says.

**Comment by rstrouse** [comment]
Hold on.  Check the country code for pi configuration and make sure it is set appropriately.

**Reply by rock-crusher** [reply]
Ok, so I went back and tried to access the regional settings, but all I got were errors!
![1  terminal](https://user-images.githubusercontent.com/94593954/234107652-e7940ab4-acd9-4de8-8f6c-7f7d80713a89.png)
![2  Localisation Options](https://user-images.githubusercontent.com/94593954/234107757-756a1bee-0157-4fd7-a2f8-f5433e40f322.png)
![3  L1 Local Configure](https://user-images.githubusercontent.com/94593954/234107799-f5ad6f17-250b-4b3a-881c-dff246cd9310.png)
![4  Error L1 Locale](https://user-images.githubusercontent.com/94593954/234107844-d1ead2ae-82ea-4ea2-bcd5-ce65afe87aaf.png)
![5  L2 Timezone     Configure time zone](https://user-images.githubusercontent.com/94593954/234107905-2d3e2c10-b758-47c8-a501-10fca1035b4c.png)
![6  There was an error running option L2 Timezone](https://user-images.githubusercontent.com/94593954/234107956-b7ba29bc-eace-4e61-a785-19e34cc91607.png)
![7  L4 Wlan Country Set](https://user-images.githubusercontent.com/94593954/234108000-60b1d04a-e16f-47f4-ad1f-98602aeb1f44.png)
![8  error running option L4 WLAN Country ](https://user-images.githubusercontent.com/94593954/234108050-e0df951d-542c-4dbd-aa7b-89015119232c.png)
Is there a file somewhere that can be accessed to manipulate the settings, maybe it somehow became corrupt?

**Comment by rstrouse** [comment]
I think the file is `/etc/wpa_supplicant/wpa_supplicant.conf`.  So obvious... said no one.

**Reply by rock-crusher** [reply]
lol… do you know where in the pi folders/files that would be located?
Hell, even if I knew where it was, I wouldn’t know what to do with it anyway 😜

**Reply by rstrouse** [reply]
From a console type `sudo nano /etc/wpa_supplicant/wpa_supplicant.conf` and it should bring up the file.

If you post its contents here maybe between us amateurs' we can make someone dangerous enough to figure it out.  You can remove the psk text from the pasted file when you paste it here.

**Reply by rock-crusher** [reply]
<img width="491" alt="GNU nano 3 2" src="https://user-images.githubusercontent.com/94593954/234156037-aa4d10e4-1bc9-45d5-a40b-6b0007dbaccb.png">

**Reply by rstrouse** [reply]
Is there an extra line at the top of that file?  If there is remove it.  Also please run the following command
`ls /var/run/wpa_supplicant` and post the results here.

**Reply by rock-crusher** [reply]
Not sure what you meant by: 
“Is there an extra line at the top of that file? If there is remove it.”
So I went ahead anyway and ran the following command:
ls /var/run/wpa_supplicant
This is the result:
![No such file or directory](https://user-images.githubusercontent.com/94593954/234389598-21350769-54d7-47e1-97ec-fbab460f81a0.png)

**Reply by rstrouse** [reply]
What version of Raspberry pi is this?  That should be part of the OS.

Type `groups pi` and make sure the pi user is part of the `netdev` group.

**Reply by rock-crusher** [reply]
Not sure if this is what you need, or not.

This is becoming way too complicated for me and time consuming.

I think I’m going to once again start all over from scratch, plug in a new card and muddle my way through the reloading of everything.

Not looking forward to it.  🤢

As always rstrouse, 

Thanks again for all your help!  

![version](https://user-images.githubusercontent.com/94593954/234505713-2a522236-e70f-4e15-b651-7c024267343f.png)

![groups](https://user-images.githubusercontent.com/94593954/234505798-b7d2499b-0888-4b22-b35b-8fc8f9425540.png)

**Reply by rstrouse** [reply]
Something in the OS is not correct.  It looks like the network configuration folder is empty.

**Comment by rock-crusher** [comment]
@rstrouse 
Once I get the new OS up and running do you know what files I could simply copy/paste from the old SD card to the new one so I don’t have to go through the entire programming/scheduling process etc.?

**Reply by rstrouse** [reply]
I do.  I was going to leave it at that but I suppose you want me to tell you.  Put simply they are the ones in the data directory.  That includes REM as well.
